core/src/test/java/org/mule/processor/IdempotentRedeliveryPolicyTestCase.java

```diff
@@ -90,6 +86,7 @@ public MuleEvent answer(InvocationOnMock invocationOnMock) throws Throwable
         muleLockFactory.initialise();
         when(mockMuleContext.getLockFactory()).thenReturn(muleLockFactory);
         when(mockMuleContext.getRegistry().get(MuleProperties.OBJECT_STORE_MANAGER)).thenReturn(mockObjectStoreManager);
+        when(mockMuleContext.getConfiguration().getDefaultEncoding()).thenReturn(UTF_8);
```

Comment:
When the transform method is executed, the encoding is determined. Since muleContext was null before, it was determined by the mule.encoding system property instead (see SystemUtils getDefaultEncoding method).